### DISCUSSION TITLE
could you make a nixos version?

### DISCUSSION THREAD

**Discussion by karomba** [JaKooLit/Hyprland-Dots#331] [Q&A] [answered]
i know nix os isnt that popular but i would love to use your dotfiles on nixos because i think it is the best looking and the most simplest

**Comment by JaKooLit** [comment]
hmmmm... how can I explain this..

I do use these dots on my nixos

and I have added also on FAQ
https://github.com/JaKooLit/Hyprland-Dots/wiki/Other_Distros#-will-this-work-on-nixos

However, my NIXOS is not written in nix language. I still use the ~/.config paths.. apart from some small things of course...

**Comment by JaKooLit** [comment]
if you want a nix-style these configs by @JohnRTitor should help you

https://github.com/JohnRTitor/nix-conf

**Comment by JohnRTitor** [comment]
[Hyprland.nix](https://github.com/JohnRTitor/nix-conf/blob/hyprland/system/hyprland.nix) has all the required packages to get started with these dots. 

You also need [envfs enabled](https://github.com/JohnRTitor/nix-conf/blob/5e1b86abea14baec770fa88e6c18f49245affa4e/system/services/default.nix#L90) for the installation to work.

My way of using these dots is like:

`./copy.sh`, answer questions and then relogin.

You'll need to get the [themes](https://github.com/JaKooLit/GTK-themes-icons) as well.

**Comment by karomba** [comment]
thank you man you were very fast

**Comment by karomba** [comment]
could you make a video on how to install these dotfiles on nix because im very new to it and would like to use hyprland on it

**Reply by JohnRTitor** [reply]
I'd recommend checking out Vimjoyer and Librephoenix's videos. They helped me a lot when starting out with NixOS. 

You'd need to configure [Flakes setup](https://nixos-and-flakes.thiscute.world/nixos-with-flakes/nixos-with-flakes-enabled). You can copy my configuration and start from there. But you'll need to [modify it a bit](https://github.com/JohnRTitor/nix-conf/blob/hyprland/preferences.nix) as per your requirement.

**Comment by dwilliam62** [comment]
There is another option.  If you are just looking for a working Hyprland configuration under NIXOS
You can use this:  https://gitlab.com/Zaney/zaneyos   His config uses flakes, and home manager for an integrated approach supporting configurations for multiple hosts
and you can fork his project to modify and save your hosts in your own git repo. 

You start with a basic NIXOS install,  no desktop needed.  You can use the GNOME Nixos live cd to build that base image.

Then you would install the ZaneyOS on top of that.   It does not use JaKool dots.  On an earlier version of zaneyos I did manually 
layer in his config.   I have not tried Jakool's  with the current version of ZaneyOS   

Lastly a new version of Zaneyos is due soon with some improvements to the UI

 
![image](https://github.com/JaKooLit/Hyprland-Dots/assets/129223418/b25d66e6-d102-4c71-a71e-c8ada4bb395f)

**Comment by karomba** [comment]
i dont know how to do this stuff so im either gonna stay on gnome or install a different distro

**Reply by dwilliam62** [reply]
Up to you.  The zaneyos is a good choice if you want to try hyprland with NIXOS   I am new to NixOS  ~6mos and this is what I use for NIXOS   Or try it on a spare machine?  Old laptop?

**Reply by karomba** [reply]
i dont have a spare machine so i cant really test it and i just installed nixos like 3 days ago so i am very new as i said but thanks for the recommendation

**Comment by JaKooLit** [comment]
https://github.com/JaKooLit/NixOS-configs

here are my nixos configs

and then I just clone the Hyprland-Dots

and run copy.sh  

did some small adjustment on Startup_Apps.conf in ~/.config/hypr/UserConfigs/

and then thats how I use it

you need to setup flakes though as it needs for wallust dev branch

**Reply by karomba** [reply]
so i basicly git clone it and then build the config and copy your dotfiles?

**Comment by JaKooLit** [comment]
git clone the Hyprland-Dots yes

but you need to setup nixos yourself

but remember on this note
https://github.com/JaKooLit/Hyprland-Dots/wiki/Other_Distros#-will-this-work-on-nixos

plus you need to setup flakes for wallust dev

**Comment by JaKooLit** [comment, marked as answer]
not exactly written on nix but heck it works :)

https://github.com/JaKooLit/NixOS-Hyprland
